app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Validator.scala

```diff
@@ -44,42 +41,42 @@ case object Validator {
 
   }
 
-  implicit val format = OFormat(reads, writes)
+  implicit val format: OFormat[Validator] = OFormat(reads, writes)
 
 }
 
 case class HMRCUTRPostcodeCheckValidator(errorMessage: String, utr: FormCtx, postcode: FormCtx) extends Validator
 
 object HMRCUTRPostcodeCheckValidator {
-  private val basic: OFormat[HMRCUTRPostcodeCheckValidator] = Json.format[HMRCUTRPostcodeCheckValidator]
-  private val writesCustom: OWrites[HMRCUTRPostcodeCheckValidator] = OWrites { o =>
+  private val basic: OFormat[HMRCUTRPostcodeCheckValidator] = derived.oformat
+  private val writes: OWrites[HMRCUTRPostcodeCheckValidator] = OWrites { o =>
     Json.obj("validatorName" -> "hmrcUTRPostcodeCheck") ++
       basic.writes(o)
   }
 
-  private val writes: OWrites[HMRCUTRPostcodeCheckValidator] = writesCustom
-  private val readCustom: Reads[HMRCUTRPostcodeCheckValidator] = ((JsPath \ "errorMessage").read[String] and
-    (JsPath \ "parameters" \\ "utr").read[FormCtx] and
-    (JsPath \ "parameters" \\ "postcode").read[FormCtx])(HMRCUTRPostcodeCheckValidator.apply _)
+  private val readCustom: Reads[HMRCUTRPostcodeCheckValidator] =
+    ((JsPath \ "errorMessage").read[String] and
+      (JsPath \ "parameters" \ "utr").read(FormCtx.readsForTemplateJson) and
+      (JsPath \ "parameters" \ "postcode").read(FormCtx.readsForTemplateJson))(HMRCUTRPostcodeCheckValidator.apply _)
+  private val reads = (basic: Reads[HMRCUTRPostcodeCheckValidator]) | readCustom
 
-  private val reads = readCustom | (basic: Reads[HMRCUTRPostcodeCheckValidator])
-  implicit val format: OFormat[HMRCUTRPostcodeCheckValidator] = OFormat(reads, writesCustom)
+  implicit val format: OFormat[HMRCUTRPostcodeCheckValidator] = OFormat(reads, writes)
 }
 
 case class BankAccoutnModulusCheck(errorMessage: String, accountNumber: FormCtx, sortCode: FormCtx) extends Validator
 
 object BankAccoutnModulusCheck {
-  private val basic: OFormat[BankAccoutnModulusCheck] = Json.format[BankAccoutnModulusCheck]
-  private val writesCustom: OWrites[BankAccoutnModulusCheck] = OWrites { o =>
+  private val basic: OFormat[BankAccoutnModulusCheck] = derived.oformat
+  private val writes: OWrites[BankAccoutnModulusCheck] = OWrites { o =>
```

Comment:
I'd get rid of this custom `writes` and used `basic` in its place. See comment above.